pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go

```diff
@@ -209,16 +219,50 @@ install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~
 
 .PHONY: uninstall
 uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
-	$(KUSTOMIZE) build config/crd | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
+	$(KUSTOMIZE) build config/crd | $(KUBECTL) delete -n $(NAMESPACE) --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
-	$(KUSTOMIZE) build config/default | $(KUBECTL) apply -f -
+	$(KUSTOMIZE) build config/default | $(KUBECTL) -n $(NAMESPACE) apply -f -
+	$(KUBECTL) wait deployment.apps/{{ .ProjectName }}-controller-manager --for condition=Available --namespace $(NAMESPACE) --timeout 5m
 
 .PHONY: undeploy
 undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
-	$(KUSTOMIZE) build config/default | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
+	$(KUSTOMIZE) build config/default | $(KUBECTL) delete -n $(NAMESPACE) --ignore-not-found=$(ignore-not-found) -f -
+
+.PHONY: redeploy
+redeploy: deploy ## Redeploy controller with new docker image.
+	$(KUBECTL) rollout restart -n $(NAMESPACE) deploy/{{ .ProjectName }}-controller-manager
+
+.PHONY: kind-load
+kind-load: docker-build kind ## Build and upload docker image to the local Kind cluster.
+	$(KIND) load docker-image ${IMG} --name $(KIND_CLUSTER_NAME)
```

Comment:
I do not think we should add targets which are specific to any vendor in the default scaffolds.
Users are able to do that if they wish, and many projects are distributed without a specific vendor.

Reply:
Let's discuss vendor(kind)-)-related issue in a certain thread, to not spread this discussion over all threads, I'd suggest the [first one](https://github.com/kubernetes-sigs/kubebuilder/pull/3989#discussion_r1649181013).